### DISCUSSION TITLE
How to map autoroute field to an integer, I need the permalink URL to populate with integer and not string.

### DISCUSSION THREAD

**Discussion by Anchaljoshi** [OrchardCMS/OrchardCore#13232] [Q&A]
Hi, I need to map autoroute to Integer value, and while publishing we need autoroute URL to be generated with an auto incremented integer(if we leave the field empty it will automatically generate the URL) and not with string value, kindly assist.

 at present the link is generated as 

localhost:1234/titleofthecontent

I need the URL to look like 'localhost:1234/12(Integer)

**Comment by Piedone** [comment]
As the Autoroute pattern under the content type, part settings, you can use `{{ Model.ContentItem.Id }}` ([see docs](https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#modelcontentitem)) to get the numerical ID of the content item. This will be a unique, and always incrementing integer, but it won't necessarily be consecutive (because these IDs are shared among all content items). You won't be able to safely export and import items either, if you want to keep generating new IDs, because then those might clash.

If you want consecutive IDs, you can implement your own ID generation logic and add a Liquid tag for it, and use it as the Autoroute pattern.

All this being said, I'd generally recommend against all of this and suggest just using human-readable permalinks.

**Reply by Anchaljoshi** [reply]
Thank you for your quick reply, however when I am binding autoroute setting to Model.ContentItem.Id, It is creating URL as 0_1 and so on, if we want consecutive IDs, will we need to customize the code as there is no setting or mapping on liquid field on UI, can we extend the model to include ID from new table.
